docs/zkapps/tutorials/09-recursion.mdx

```diff
@@ -0,0 +1,312 @@
+---
+title: 'Tutorial 9: Recursion'
+hide_title: true
+sidebar_label: 'Tutorial 9: Recursion'
+---
+
+:::info
+
+Please note that zkApp programmability is not yet available on Mina Mainnet, but
+zkApps can now be deployed to Berkeley Testnet.
+
+:::
+
+# Tutorial 9: Recursion
+
+## Overview
+
+One of the most powerful features of zkApps is recursion.
+
+With recursion, you can realize composability between zero knowledge proofs. This unlocks many powerful technical abilities, such as creating high-throughput applications, creating proofs of large computations, and constructing multi-party proofs.
+
+### Scaling Throughput - zkRollups & App Chains
+
+Verifying large amounts of information is usually challenging for blockchains. But with zero knowledge proofs, and particularly recursive ZKPs, this becomes far easier.
+
+By leveraging recursive verification, it is possible to easily construct both zkRollups and App Chains. We’ll give an example of a simple zkRollup near the end of this tutorial.
+
+Recursive composition even gives us flexibility to handle live demand. Recursive composition gives you flexibility to choose between a high tree height (higher throughput, but logarithmically slower latency), and lower tree height (lower throughput, but faster latency). The depth of the tree can be modified live to handle whatever traffic actually present on the network, while still offering optimal latency to commit transactions back to the chain.
+
+For an example of an ‘app chain’, one could imagine an on-chain trading pair that uses an orderbook. Rolling up the transactions for the application with zero knowledge proofs, can allow our app to handle the expensive computations of keeping buy and sell orders sorted, while still posting full verification to the chain.
+
+At the end of this tutorial, we’ll review a simple zkRollup example that can be used for implementing a zkRollup or an app chain.
+
+### Scaling Proof Size
+
+Recursive ZKPs also give you the ability to construct very large transactions that wouldn’t otherwise be possible.
+
+For example, recursive ZKPs could be used to prove the output of a machine learning (ML) model. This could be used to prove an inference is truly generated by a model, or for a more computationally intensive case, to verify a model has been trained on a particular dataset.
+
+### Off-chain, multi-party proof construction
+
+Recursive zero knowledge proofs also make it easy to allow multiple parties to construct transactions. To accomplish off-chain proof construction, one or more parties can recursively update a ZKP and its associated public state. That state can then be sent as part of a transaction on-chain, or used as part of an off-chain application leveraging ZKPs.
```

Comment:
Would it be more clear to append to the follwing to the last sentence?: "When the multi-party aspect is completed, the state can then be sent..."